File: implementations/node/sdk/src/contracts/pipelines/pipeline-step-definition.ts
import type { AiSdkJsonValue } from "../../json/json-value.js";
import type { AiSdkInvocationDefinition } from "./invocation-definition.js";
import type { AiSdkPipelineStepExecutionDefinition } from "./pipeline-step-execution-definition.js";

export interface AiSdkPipelineStepDefinition {
  readonly name: string;
  readonly stepKey: string;
  readonly executionLanguage?: string;
  readonly invocation?: AiSdkInvocationDefinition;
  readonly order: number;
  readonly dependsOn?: readonly string[];
  readonly input?: Readonly<Record<string, AiSdkJsonValue>>;
  readonly config?: Readonly<Record<string, AiSdkJsonValue>>;
  readonly execution?: AiSdkPipelineStepExecutionDefinition;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/pipelines/pipeline_step_definition.py
from __future__ import annotations

from dataclasses import dataclass, field

from ...json_types import AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from .invocation_definition import AiSdkInvocationDefinition
from .pipeline_step_execution_definition import AiSdkPipelineStepExecutionDefinition


@dataclass(frozen=True)
class AiSdkPipelineStepDefinition(AiSdkWireModel):
    name: str
    step_key: str
    order: int
    execution_language: str | None = None
    invocation: AiSdkInvocationDefinition | None = None
    depends_on: tuple[str, ...] = ()
    input: dict[str, AiSdkJsonValue] = field(default_factory=dict, metadata={"json": True})
    config: dict[str, AiSdkJsonValue] = field(default_factory=dict, metadata={"json": True})
    execution: AiSdkPipelineStepExecutionDefinition | None = None
